sisl/geom/nanoribbon.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+
+from sisl import geom, Atom, Geometry
+
+__all__ = ['nanoribbon']
+
+
+def nanoribbon(width, kind='armchair', bond=1.42, atom=None, sat_bond=1.09, sat_atom=None):
```

Comment:
I would prefer the order `bond, atom, width, kind='armchair', saturate/saturation=None`.

Then I would do  
`saturation` may be either an `Atom` or `tuple(Atom, bond)`
If only an atom I would read the `calculated` radius from the `PeriodicTable` for the nano-ribbon atom and the saturation atom and use that as the bond.
Something like:
```python
sat_bond = sisl.PeriodicTable.radius(atom.Z) + sisl.PeriodicTable(saturation.Z)
```
For C and H it would be `1.2 A`. For a tuple, the bond length supplied will be used.

Also, since H does not really play a big role it should perhaps default to *not* add saturation atoms if `saturation` is None? I.e. for TB one would not use the hydrogens?

Reply:
Agreed, removed saturation stuff from this PR, could you in the same go update the argument order? When changed it matches better the other geometry creation tools.

Reply:
Fixed

Reply:
Well... only partially. Do you want to enforce the user to specify `bond` and `atom` parameters in addition to `width`?

Reply:
Yeah, I probably want that. This means we have a general nanoribbon, then we can add

- `graphene_nanoribbon` (and the shorter `agnr`/`zgnr`)

In this way it also highlights that the nanoribbon may be a hBN for instance.

Also for `width` I think this would be fair to force users as input.

Reply:
`nanoribbon` fixed

Reply:
Added also the more specific geometries `graphene_nanoribbon`, `agnr` and `zgnr`.